From 8fe17dbda7a767ded4df43a2d392ebfeca41c2b8 Mon Sep 17 00:00:00 2001
From: Giuseppe Lettieri <g.lettieri@iet.unipi.it>
Date: Thu, 10 Feb 2022 17:31:19 +0100
Subject: [PATCH 3/3] gdb: enable read/write to (io)apic registers

---
 system/physmem.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/system/physmem.c b/system/physmem.c
index 75389064a8..152745e950 100644
--- a/system/physmem.c
+++ b/system/physmem.c
@@ -3034,6 +3034,8 @@ static inline MemTxResult address_space_write_rom_internal(AddressSpace *as,
         if (!(memory_region_is_ram(mr) ||
               memory_region_is_romd(mr))) {
             l = memory_access_size(mr, l, addr1);
+            if (type == WRITE_DATA)
+                address_space_write(as, addr, attrs, ptr, l);
         } else {
             /* ROM/RAM case */
             ram_ptr = qemu_map_ram_ptr(mr->ram_block, addr1);
@@ -3558,8 +3560,11 @@ int cpu_memory_rw_debug(CPUState *cpu, vaddr addr,
     hwaddr phys_addr;
     vaddr l, page;
     uint8_t *buf = ptr;
+    CPUState *save_cpu;
 
     cpu_synchronize_state(cpu);
+    save_cpu = current_cpu;
+    current_cpu = cpu;
     while (len > 0) {
         int asidx;
         MemTxAttrs attrs;
@@ -3589,6 +3594,7 @@ int cpu_memory_rw_debug(CPUState *cpu, vaddr addr,
         buf += l;
         addr += l;
     }
+    current_cpu = save_cpu;
     return 0;
 }
 
-- 
2.45.2

